Add vim cursor color variable support

prurph opened the pull request:
As discussed in https://github.com/catppuccin/jupyterlab/issues/9. The vim extension allows `--jp-vim-cursor-color` to style the block cursor for vim normal mode. This PR styles that consistently with the normal line cursor that is colored by `--jp-editor-cursor-color` and which this theme already styles.

The `!important` is required because the vim extension also sets this variable to a default, so this ensures the theme's value overrides it. I've also verified that overrides in  `~/.jupyter/custom/custom.css` do correctly override this value, provided they have `!important`. Important was needed to override the vim extension setting in custom.css anyway, so this is not a change in behavior.

prurph commented:
As I mentioned in the issue, if you're not a fan of adding this to catppuccin itself, I'm happy to close this or change it to a note in the readme telling vim users how they can make this change themselves. Thanks!

gabrielmagno commented:
Hi @prurph , thanks for the PR.

I'm not against adding this workaround at all! I would be happy to make it compatible with vim mode.

Unfortunately your fix seems not to be working. I tried the same thing myself here (adding to variables.css).
Maybe yours is working because you have the custom.css? Could you check with a "clean" install of jupyter lab?

prurph commented:
Hi @gabrielmagno, thanks for checking it out! Here's what I did to verify with a fresh install. Forgive me for the wall of text/over-explanation:

### Create new environment, install jupyter, and run it to test.

- At this stage there is no theme and no vim mode.

```shell
$ mv ~/.jupyter/custom/custom.css ~/.jupyter/custom/custom.css.bak
$ poetry new cp-jup-test && cd cp-jup-test
$ poetry install jupyter
$ poetry run jupyter notebook
```

### Add vim extension

```shell
$ poetry install jupyterlab-vim
$ poetry run jupyter notebook
```

- At this stage there is no catppuccin and the block cursor is green

<img src="https://github.com/user-attachments/assets/096568c8-2590-42dd-9ada-fc3ccb1948dc" height="100">

### Install catppuccin from my fork (make sure to use the `patch-1` branch)

```shell
# Install branch patch-1 (autonamed by github when making a PR using the editor)
$ poetry add git+https://github.com/prurph/jupyterlab.git#patch-1
$ poetry run jupyter labextension

~/Library/Caches/pypoetry/virtualenvs/cp-jup-test-gDNj58vn-py3.12/share/jupyter/labextensions
        jupyterlab_pygments v0.3.0 enabled OK (python, jupyterlab_pygments)
        catppuccin_jupyterlab v0.2.2 enabled OK (python, catppuccin_jupyterlab)
        @jupyter-notebook/lab-extension v7.2.1 enabled OK
        @axlair/jupyterlab_vim v4.1.3 enabled OK (python, jupyterlab_vim)
        @jupyter-widgets/jupyterlab-manager v5.0.11 enabled OK (python, jupyterlab_widgets)

$ poetry run jupyter notebook
```

- Now there is catppuccin and the correct cursor color in both normal (block) and insert (line):

<img src="https://github.com/user-attachments/assets/e8d8dd8e-4ee3-4c99-baf4-7f7f9bd0e23c" height="150" width="225">

### Test overwriting with config:

```shell
$ echo -e ':root {\n    --jp-vim-cursor-color: var(--ctp-plt-blue) !important;\n}' > ~/.jupyter/custom/custom.css
$ poetry run jupyter notebook
```

- Now the cursor is blue, as expected

Doing this from scratch I have confirmed a few things:

- `!important` does not seem to be required for the catppuccin declaration; I have removed it from my PR. I'm a little worried the order of extensions matters here (and may be indeterminate on different setups) but with the steps above it works on my machine as seen.
- I removed it because after further testing if catppuccin uses `!important`, custom css from the user doesn't seem to override this properly. By removing it from the theme, the user can (and must) use `!important` to set the variable.

Let me know if this is working for you. One issue I ran into initially I'll share just in case:

- When I was just making changes to my local repository and `pip install -e .` it, nothing changed because I wasn't rebuilding the extension. I found I had to `npm run clean:labextension && npm run build && pip install -e .`; otherwise I just kept reinstalling the same generated extension files without ever rebuilding them to reflect my code changes.

gabrielmagno commented:
Oh, I cloned your fork but forgot to change the branch, sorry about that 😅
Now, setting the branch properly it is working fine!

I modified your branch just to add reference for the vim plugin we are referring to.

I will just wait for the CICD checks to run, merge, and create a new release with this improvement.

Thank you very much @prurph !

prurph commented:
Yay! Thank _you_ for this great Jupyter extension!
